src/audio/dai.c

```diff
@@ -600,6 +604,18 @@ static int dai_config_prepare(struct comp_dev *dev)
 		return -EIO;
 	}
 
+#if CONFIG_IMX8ULP
+	imx8ulp_dmamux_rx_int = 0;
+	imx8ulp_dmamux_tx_int = 0;
+	if (dai->dai_index == 5) {
+		imx8ulp_dmamux_rx_int = IMX8ULP_DMAMUX2_SAI5_RX;
+		imx8ulp_dmamux_tx_int = IMX8ULP_DMAMUX2_SAI5_TX;
+	} else if (dai->dai_index == 6) {
+		imx8ulp_dmamux_rx_int = IMX8ULP_DMAMUX2_SAI6_RX;
+		imx8ulp_dmamux_tx_int = IMX8ULP_DMAMUX2_SAI6_TX;
+	}
```

Comment:
how about `switch (dai->dai_index)`

Reply:
ok, sorry, I don't think this should be here at all. This is a generic file, any hardware-specific code should be avoided here. I think this can perfectly be done in `dai_config_dma_channel()` and in `edma_set_config()`. You might need private DMA channel data like `struct dw_drv_plat_data` (see e.g. sof/src/platform/intel/cavs/lib/dma.c) or maybe you don't even need this since IRQ numbers are already in `struct dma_plat_data`.